pkg/controller/servicemesh/controlplane/htpasswd.go

```diff
@@ -32,9 +32,13 @@ func (r *controlPlaneInstanceReconciler) patchHtpasswdSecret(ctx context.Context
 			log.Error(err, "failed to generate the HTPasswd password")
 			return err
 		}
-		h := sha1.New()
-		h.Write([]byte(rawPassword))
-		auth = "internal:{SHA}" + base64.StdEncoding.EncodeToString(h.Sum(nil))
+		hashedPassword, err1 := bcrypt.GenerateFromPassword([]byte(rawPassword), bcrypt.DefaultCost)
+		if err1 != nil {
+			log.Error(err1, "failed to encrypt the raw password")
+			return err1
+		}
+		username := "internal"
+		auth = fmt.Sprintf("%s:%s", username, string(hashedPassword))
```

Comment:
Don't you need to add a special prefix to the password, indicating that it's using bcrypt, or does bcrypt itself add the correct prefix?

Reply:
No, `bcrypt` does it.

Reply:
![Screenshot from 2022-11-08 13-03-18](https://user-images.githubusercontent.com/17457695/200559190-a6835c26-0622-4bca-bf68-f7e6daf96d2e.png)